gtsfm/scene_optimizer.py

```diff
@@ -367,79 +371,69 @@ def save_visualizations(
     Returns:
         A list of Delayed objects after saving the different visualizations.
     """
-    aligned_ba_input_graph = dask.delayed(lambda x, y: x.align_via_Sim3_to_poses(y))(ba_input_graph, gt_pose_graph)
-    aligned_ba_output_graph = dask.delayed(lambda x, y: x.align_via_Sim3_to_poses(y))(ba_output_graph, gt_pose_graph)
-    viz_graph_list = []
-    viz_graph_list.append(dask.delayed(viz_utils.save_sfm_data_viz)(aligned_ba_input_graph, PLOT_BA_INPUT_PATH))
-    viz_graph_list.append(dask.delayed(viz_utils.save_sfm_data_viz)(aligned_ba_output_graph, PLOT_RESULTS_PATH))
-    viz_graph_list.append(
-        dask.delayed(viz_utils.save_camera_poses_viz)(
-            aligned_ba_input_graph, aligned_ba_output_graph, gt_pose_graph, PLOT_RESULTS_PATH
-        )
-    )
-    return viz_graph_list
-
-
-def save_gtsfm_data(image_graph: List[Delayed], ba_input_graph: Delayed, ba_output_graph: Delayed) -> List[Delayed]:
-    """Saves the Gtsfm data before and after bundle adjustment.
+    aligned_ba_input = ba_input.align_via_Sim3_to_poses(gt_poses)
```

Comment:
is this duplicated when we compute metrics somewhere?

Reply:
Removed it. Thanks

Reply:
Its aligned in a different function. We do align it in the `BundleAdjustmentOptimizer::evaluate`, but that aligned `GtsfmData` objects are not returned.

Reply:
was this the reason why we were aligning ba_input in the other TODO comment? Its pretty expensive so we should only do it once I think